pkgs/development/compilers/teyjus/default.nix

```diff
@@ -11,7 +11,31 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-nz7jZ+GdF6mZQPzBrVD9K/RtoeuVRuhfs7vej4zDkhg=";
   };
 
-  patches = [ ./fix-lex-to-flex.patch ];
+  prePatch = ''
+    dos2unix source/OMakefile
+    sed -i "/TST/d" source/OMakefile
+    mv source/front/ccode_stubs.c source/front/ccode_stubs_c.c
+    mv source/front/ccode_stubs.mli source/front/ccode_stubs.ml
+    rm -r source/front/caml
+  '';
+
+  patches = [
+    ./fix-lex-to-flex.patch
+    (fetchpatch {
+      url = "https://github.com/gentoo/gentoo/raw/6c807fa38179efb2bfe09acf1a332c7bdf7fe0c0/dev-lang/teyjus/files/teyjus-2.1-p003-Removing-deprecated-function-String.set.patch";
+      sha256 = "sha256-DlEzwmcsJTwH+VAa1G4VqSAugQKPFVlBzYOi19Y56SM=";
+    })
+    (fetchpatch {
```

Comment:
Also, the upstream patch for the ccode stubs issue includes the file renamings that are currently added to `patchPhase`.